Source/Renderers/DownASTRenderable.swift

```diff
@@ -11,6 +11,7 @@ import libcmark
 
 public protocol DownASTRenderable: DownRenderable {
     func toAST(_ options: DownOptions) throws -> UnsafeMutablePointer<cmark_node>
+    func toDocument(_ options: DownOptions) throws -> Document
```

Comment:
Agreed that there's no need for this as part of the `*Renderable` protocols since it's just a convenience function

@5sw Mind making this change + documentation change below? Then everything should be good to merge